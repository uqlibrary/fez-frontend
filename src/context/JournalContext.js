import React from 'react';

export default React.createContext({
    journalDetails: null,
    jnlDisplayType: null,
});
