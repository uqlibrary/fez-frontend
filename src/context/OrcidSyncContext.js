import { createContext } from 'react';

export const OrcidSyncContext = createContext({
    orcidSyncProps: {},
});

export default OrcidSyncContext;
