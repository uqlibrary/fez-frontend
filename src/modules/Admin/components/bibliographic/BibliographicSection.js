import React from 'react';
import PropTypes from 'prop-types';

import { Section } from '../common/Section';

import { useRecordContext, useFormValuesContext } from 'context';
import { adminInterfaceConfig } from 'config/admin';
import { bibliographicParams } from 'modules/Admin/helpers';

export const BibliographicSection = ({ disabled = false }) => {
    const { record } = useRecordContext();
    const { formValues } = useFormValuesContext();

    const cards = adminInterfaceConfig[record.rek_display_type].bibliographic(bibliographicParams(record, formValues));

    return <Section cards={cards} disabled={disabled} />;
};

BibliographicSection.propTypes = {
    disabled: PropTypes.bool,
};

export default React.memo(BibliographicSection);
