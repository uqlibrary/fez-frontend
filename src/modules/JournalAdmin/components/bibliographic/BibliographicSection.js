/* istanbul ignore file */
import React, { useRef } from 'react';
import PropTypes from 'prop-types';

import { Section } from '../common/Section';

import { useJournalContext } from 'context';
import { adminInterfaceConfig } from 'config/journalAdmin';

export const BibliographicSection = ({ disabled = false }) => {
    const { jnlDisplayType } = useJournalContext();

    const cards = useRef(adminInterfaceConfig[jnlDisplayType].bibliographic());

    return <Section cards={cards.current} disabled={disabled} />;
};

BibliographicSection.propTypes = {
    disabled: PropTypes.bool,
};

export default React.memo(BibliographicSection);
