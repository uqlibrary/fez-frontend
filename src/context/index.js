import { useContext } from 'react';

import AccountContext from './AccountContext';
import EditableContext from './EditableContext';
import FormErrorsContext from './FormErrorsContext';
import FormValuesContext from './FormValuesContext';
import LocallyStoredReducerContext from './LocallyStoredReducerContext';
import OrcidSyncContext from './OrcidSyncContext';
import RecordContext from './RecordContext';
import RecordsSelectorContext from './RecordsSelectorContext';
import TabbedContext from './TabbedContext';
import ScrollToSectionContext from './ScrollToSectionContext';
import JournalContext from './JournalContext';

export const useAccountContext = () => useContext(AccountContext);
export const useEditableContext = () => useContext(EditableContext);
export const useFormErrorsContext = () => useContext(FormErrorsContext);
export const useFormValuesContext = () => useContext(FormValuesContext);
export const useLocallyStoredReducerContext = () => useContext(LocallyStoredReducerContext);
export const useOrcidSyncContext = () => useContext(OrcidSyncContext);
export const useRecordContext = () => useContext(RecordContext);
export const useRecordsSelectorContext = () => useContext(RecordsSelectorContext);
export const useTabbedContext = () => useContext(TabbedContext);
export const useScrollToSectionContext = () => useContext(ScrollToSectionContext);
export const useJournalContext = () => useContext(JournalContext);

export {
    AccountContext,
    EditableContext,
    FormErrorsContext,
    FormValuesContext,
    LocallyStoredReducerContext,
    OrcidSyncContext,
    RecordContext,
    RecordsSelectorContext,
    ScrollToSectionContext,
    TabbedContext,
    JournalContext,
};
