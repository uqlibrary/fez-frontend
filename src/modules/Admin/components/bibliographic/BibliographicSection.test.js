import BibliographicSection from './BibliographicSection';

jest.mock('../../../../context');
import { useRecordContext, useFormValuesContext } from 'context';

function setup(testProps = {}, args = { isShallow: true }) {
    const props = {
        ...testProps,
    };

    return renderComponent(BibliographicSection, props, args);
}

describe('BibliographicSection component', () => {
    it('should render default view', () => {
        useRecordContext.mockImplementation(() => ({
            record: {
                rek_pid: 'UQ:123456',
                rek_object_type_lookup: 'Record',
                fez_record_search_key_ismemberof: [
                    {
                        rek_ismemberof: 'Test collection',
                        parent: {
                            rek_security_policy: 2,
                            rek_datastream_policy: 1,
                        },
                    },
                ],
                rek_display_type: 179,
                fez_record_search_key_language: [{ rek_language: 'eng' }],
            },
        }));
        useFormValuesContext.mockImplementation(() => ({
            formValues: {
                languages: ['eng'],
            },
        }));

        const render = setup();
        expect(render.getRenderOutput()).toMatchSnapshot();
    });

    it('should render disabled view', () => {
        useRecordContext.mockImplementation(() => ({
            record: {
                rek_pid: 'UQ:123456',
                rek_object_type_lookup: 'Record',
                fez_record_search_key_ismemberof: [
                    {
                        rek_ismemberof: 'Test collection',
                        parent: {
                            rek_security_policy: 2,
                            rek_datastream_policy: 1,
                        },
                    },
                ],
                rek_display_type: 179,
                fez_record_search_key_language: [{ rek_language: 'eng' }],
            },
        }));

        useFormValuesContext.mockImplementation(() => ({
            formValues: {
                languages: ['eng'],
            },
        }));

        const render = setup({ disabled: true });
        expect(render.getRenderOutput()).toMatchSnapshot();
    });

    it('should render design form fields', () => {
        useRecordContext.mockImplementation(() => ({
            record: {
                rek_pid: 'UQ:123456',
                rek_object_type_lookup: 'Record',
                fez_record_search_key_ismemberof: [
                    {
                        rek_ismemberof: 'Test collection',
                        parent: {
                            rek_security_policy: 2,
                            rek_datastream_policy: 1,
                        },
                    },
                ],
                rek_display_type: 313,
                rek_subtype: 'Creative Work - Design/Architectural',
            },
        }));

        const render = setup();
        expect(render.getRenderOutput()).toMatchSnapshot();
    });
});
