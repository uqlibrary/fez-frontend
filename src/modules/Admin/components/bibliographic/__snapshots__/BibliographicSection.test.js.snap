// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BibliographicSection component should render default view 1`] = `
<Section
  cards={
    [
      {
        "groups": [
          [
            "rek_title",
          ],
        ],
        "title": "Title",
      },
      {
        "groups": [
          [
            "languages",
          ],
          [
            "fez_record_search_key_journal_name",
            "fez_matched_journals",
          ],
        ],
        "title": "Language of work & Journal name",
      },
      {
        "groups": [
          [
            "fez_record_search_key_isbn",
          ],
        ],
        "title": "ISBN",
      },
      {
        "groups": [
          [
            "issns",
          ],
        ],
        "title": "ISSN",
      },
      {
        "groups": [
          [
            "fez_record_search_key_place_of_publication",
            "fez_record_search_key_publisher",
          ],
          [
            "fez_record_search_key_volume_number",
            "fez_record_search_key_issue_number",
            "fez_record_search_key_article_number",
          ],
          [
            "fez_record_search_key_start_page",
            "fez_record_search_key_end_page",
            "fez_record_search_key_total_pages",
          ],
          [
            "rek_date",
          ],
          [
            "fez_record_search_key_date_available",
          ],
          [
            "rek_description",
          ],
        ],
        "title": "Bibliographic",
      },
      {
        "groups": [
          [
            "fez_record_search_key_keywords",
          ],
        ],
        "title": "Keyword(s)",
      },
      {
        "groups": [
          [
            "subjects",
          ],
        ],
        "title": "Subject",
      },
      {
        "groups": [
          [
            "fez_record_search_key_isderivationof",
          ],
        ],
        "title": "Related publications",
      },
    ]
  }
  disabled={false}
/>
`;

exports[`BibliographicSection component should render design form fields 1`] = `
<Section
  cards={
    [
      {
        "groups": [
          [
            "rek_title",
          ],
        ],
        "title": "Title",
      },
      {
        "groups": [
          [
            "languages",
          ],
        ],
        "title": "Language of work",
      },
      {
        "groups": [
          [
            "fez_record_search_key_isbn",
          ],
        ],
        "title": "ISBN",
      },
      {
        "groups": [
          [
            "issns",
          ],
        ],
        "title": "ISSN",
      },
      {
        "groups": [
          [
            "fez_record_search_key_place_of_publication",
            "fez_record_search_key_publisher",
          ],
          [
            "fez_record_search_key_edition",
            "fez_record_search_key_volume_number",
            "fez_record_search_key_issue_number",
          ],
          [
            "fez_record_search_key_series",
          ],
          [
            "fez_record_search_key_start_page",
            "fez_record_search_key_end_page",
            "fez_record_search_key_total_pages",
          ],
          [
            "rek_date",
          ],
          [
            "fez_record_search_key_date_available",
          ],
          [
            "rek_description",
          ],
          [
            "fez_record_search_key_original_format",
          ],
        ],
        "title": "Bibliographic",
      },
      {
        "groups": [
          [
            "fez_record_search_key_keywords",
          ],
        ],
        "title": "Keyword(s)",
      },
      {
        "groups": [
          [
            "subjects",
          ],
        ],
        "title": "Subject",
      },
      {
        "groups": [
          [
            "fez_record_search_key_isderivationof",
          ],
        ],
        "title": "Related publications",
      },
    ]
  }
  disabled={false}
/>
`;

exports[`BibliographicSection component should render disabled view 1`] = `
<Section
  cards={
    [
      {
        "groups": [
          [
            "rek_title",
          ],
        ],
        "title": "Title",
      },
      {
        "groups": [
          [
            "languages",
          ],
          [
            "fez_record_search_key_journal_name",
            "fez_matched_journals",
          ],
        ],
        "title": "Language of work & Journal name",
      },
      {
        "groups": [
          [
            "fez_record_search_key_isbn",
          ],
        ],
        "title": "ISBN",
      },
      {
        "groups": [
          [
            "issns",
          ],
        ],
        "title": "ISSN",
      },
      {
        "groups": [
          [
            "fez_record_search_key_place_of_publication",
            "fez_record_search_key_publisher",
          ],
          [
            "fez_record_search_key_volume_number",
            "fez_record_search_key_issue_number",
            "fez_record_search_key_article_number",
          ],
          [
            "fez_record_search_key_start_page",
            "fez_record_search_key_end_page",
            "fez_record_search_key_total_pages",
          ],
          [
            "rek_date",
          ],
          [
            "fez_record_search_key_date_available",
          ],
          [
            "rek_description",
          ],
        ],
        "title": "Bibliographic",
      },
      {
        "groups": [
          [
            "fez_record_search_key_keywords",
          ],
        ],
        "title": "Keyword(s)",
      },
      {
        "groups": [
          [
            "subjects",
          ],
        ],
        "title": "Subject",
      },
      {
        "groups": [
          [
            "fez_record_search_key_isderivationof",
          ],
        ],
        "title": "Related publications",
      },
    ]
  }
  disabled={true}
/>
`;
